src/cpp/src/whisper_pipeline_static.cpp

```diff
@@ -288,49 +278,81 @@ std::vector<int32_t> prepare_init_ids(ov::Tensor& encoder_hidden_state,
+    auto handle = std::make_shared<ov::genai::GenerationHandleImpl>(sequence_group->get_generation_stream(),
+                                                                    sequence_group->get_sampling_parameters());
+
+    auto logits = decode(encoder_hidden_state, models.decoder, init_ids, raw_metrics);
+    process_whisper_logits(logits, config, return_timestamps, {});
+
+    // sample last token only
+    int64_t output_sequence_len = logits.get_shape().at(1);
+    sequence_group->schedule_tokens(sequence_group->get_prompt_len());
+    sequence_group->set_output_seq_len(output_sequence_len);
+
+    sampler.sample({sequence_group}, logits);
+    stream_generated_tokens(streamer, handle, return_timestamps);
 
     prepare_decoder_with_past(models.decoder_with_past, models.decoder, init_ids.size());
 
-    for (size_t i = 0; i < max_new_tokens - 1; i++) {
-        auto output_token = decode_with_past(models.decoder_with_past,
-                                             output_tokens.back(),
-                                             i + init_ids.size(),
-                                             config,
-                                             raw_metrics,
-                                             return_timestamps,
-                                             output_tokens);
-        update_past_key_value(models.decoder_with_past, models.decoder_with_past, i + init_ids.size());
-
-        if (output_token == config.eos_token_id) {
-            break;
-        }
+    while (!sequence_group->has_finished() && !sequence_group->handle_dropped()) {
+        sequence_group->schedule_tokens(1);
+        const auto running_sequences = sequence_group->get_running_sequences();
+        OPENVINO_ASSERT(running_sequences.size() == 1u);
+        auto last_token = running_sequences.front()->get_generated_ids().back();
+        auto last_idx = running_sequences.front()->get_generated_ids().size();
+
+        auto logits = decode_with_past(models.decoder_with_past,
+                                       last_token,
+                                       last_idx + init_ids.size(),
+                                       raw_metrics);
+        process_whisper_logits(logits, config, return_timestamps, running_sequences.front()->get_generated_ids());
+        update_past_key_value(models.decoder_with_past, models.decoder_with_past, last_idx + init_ids.size());
+
+        sampler.sample({sequence_group}, logits);
+        stream_generated_tokens(streamer, handle, return_timestamps);
+    }
 
-        output_tokens.push_back(output_token);
+    ov::genai::EncodedResults results;
+    // NB: Only batch=1 is supported now
+    results.scores.resize(1u);
+    results.scores[0] = 0u;
+    results.tokens.resize(1u);
 
-        if (!return_timestamps && streamer && streamer->put(output_token)) {
-            return {true, output_tokens};
-        }
-    }
+    OPENVINO_ASSERT(sequence_group->get_finished_sequences().size() == 1u);
```

Comment:
What situation can that there will be more than one sequence here?

Reply:
Such cases not supported for NPU yet